client/my-sites/current-site/index.jsx

```diff
@@ -92,12 +93,10 @@ class CurrentSite extends Component {
 	render() {
 		const { selectedSite, translate, anySiteSelected, rtlOn } = this.props;
 
-		if ( ! anySiteSelected.length ) {
+		if ( ! anySiteSelected.length || ( ! selectedSite && ! this.props.hasAllSitesList ) ) {
```

Comment:
Instead of this condition, I think the `getSelectedOrAllSites` should return `null` if no site is selected and when it doesn't have the full site list. Now, when the full site list is not available, the selector will return an incomplete list. That's a wrong result and you need to guard against it.